Unresolved External Symbols with Node 12

I've updated my nodejs installation to v12.2.0, and now I get a huge error while compiling an extension that worked fine previously.

```
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) protected: static class v8::internal::Object * * __cdecl v8::HandleScope::CreateHandle(class v8::internal::Isolate *,class v8::internal::Object *)" (__imp_?CreateHandle@HandleScope@v8@@KAPEAPEAVObject@internal@2@PEAVIsolate@42@PEAV342@@Z) referenced in function "void __cdecl neon::complete_task(struct uv_work_s *)" (?complete_task@neon@@YAXPEAUuv_work_s@@@Z)
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) private: class v8::internal::Object * * __cdecl v8::EscapableHandleScope::Escape(class v8::internal::Object * *)" (__imp_?Escape@EscapableHandleScope@v8@@AEAAPEAPEAVObject@internal@2@PEAPEAV342@@Z) referenced in function Neon_Convert_ToObject
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: int __cdecl v8::String::Utf8Length(void)const " (__imp_?Utf8Length@String@v8@@QEBAHXZ) referenced in function Neon_String_Utf8Length
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class v8::Local<class v8::Function> __cdecl v8::FunctionTemplate::GetFunction(void)" (__imp_?GetFunction@FunctionTemplate@v8@@QEAA?AV?$Local@VFunction@v8@@@2@XZ) referenced in function Neon_Class_Constructor
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) private: static class v8::internal::Object * * __cdecl v8::V8::GlobalizeReference(class v8::internal::Isolate *,class v8::internal::Object * *)" (__imp_?GlobalizeReference@V8@v8@@CAPEAPEAVObject@internal@2@PEAVIsolate@42@PEAPEAV342@@Z) referenced in function "public: virtual void __cdecl neon::BaseClassMetadata::construct(class v8::FunctionCallbackInfo<class v8::Value> const &)" (?construct@BaseClassMetadata@neon@@UEAAXAEBV?$FunctionCallbackInfo@VValue@v8@@@v8@@@Z)
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) private: static void __cdecl v8::V8::DisposeGlobal(class v8::internal::Object * *)" (__imp_?DisposeGlobal@V8@v8@@CAXPEAPEAVObject@internal@2@@Z) referenced in function "protected: virtual __cdecl neon::ClassMetadata::~ClassMetadata(void)" (??1ClassMetadata@neon@@MEAA@XZ)
          libneon_runtime-388232a462956889.rlib(neon.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) private: static void __cdecl v8::V8::MakeWeak(class v8::internal::Object * *,void *,void (__cdecl*)(class v8::WeakCallbackInfo<void> const &),enum v8::WeakCallbackType)" (__imp_?MakeWeak@V8@v8@@CAXPEAPEAVObject@internal@2@PEAXP6AXAEBV?$WeakCallbackInfo@X@2@@ZW4WeakCallbackType@2@@Z) referenced in function "public: virtual void __cdecl neon::BaseClassMetadata::construct(class v8::FunctionCallbackInfo<class v8::Value> const &)" (?construct@BaseClassMetadata@neon@@UEAAXAEBV?$FunctionCallbackInfo@VValue@v8@@@v8@@@Z)
```

I assume this is because of some deprecated functionality, or maybe a header just needs updating. I'm not quite sure. Could be related to #202 ? 

If it's relevant, I'm compiling on Windows 10, with MSVC 2019.

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,6 +19,7 @@ matrix:
 
 env:
   matrix:
+    - TRAVIS_NODE_VERSION="12"
     - TRAVIS_NODE_VERSION="10"
     - TRAVIS_NODE_VERSION="8"
     - TRAVIS_NODE_VERSION="6"

--- a/crates/neon-runtime/package.json
+++ b/crates/neon-runtime/package.json
@@ -10,6 +10,6 @@
   "dependencies": {
     "bindings": "1.2.1",
     "node-gyp": "^3.5.0",
-    "nan": "^2.10.0"
+    "nan": "^2.14.0"
   }
 }

--- a/crates/neon-runtime/src/neon.cc
+++ b/crates/neon-runtime/src/neon.cc
@@ -160,7 +160,11 @@ extern "C" bool Neon_String_New(v8::Local<v8::String> *out, v8::Isolate *isolate
 }
 
 extern "C" int32_t Neon_String_Utf8Length(v8::Local<v8::String> str) {
-  return str->Utf8Length();
+  #if NODE_MODULE_VERSION >= NODE_11_0_MODULE_VERSION
+    return str->Utf8Length(v8::Isolate::GetCurrent());
+  #else
+    return str->Utf8Length();
+  #endif
 }
 
 extern "C" size_t Neon_String_Data(char *out, size_t len, v8::Local<v8::Value> str) {
@@ -338,7 +342,7 @@ extern "C" void *Neon_Class_GetAllocateKernel(v8::Local<v8::External> wrapper) {
 }
 
 extern "C" bool Neon_Class_Constructor(v8::Local<v8::Function> *out, v8::Local<v8::FunctionTemplate> ft) {
-  v8::MaybeLocal<v8::Function> maybe = ft->GetFunction();
+  v8::MaybeLocal<v8::Function> maybe = Nan::GetFunction(ft);
   return maybe.ToLocal(out);
 }
 
@@ -394,7 +398,7 @@ extern "C" bool Neon_Class_AddMethod(v8::Isolate *isolate, void *metadata_pointe
 
 extern "C" bool Neon_Class_MetadataToConstructor(v8::Local<v8::Function> *out, v8::Isolate *isolate, void *metadata) {
   v8::Local<v8::FunctionTemplate> ft = static_cast<neon::ClassMetadata *>(metadata)->GetTemplate(isolate);
-  v8::MaybeLocal<v8::Function> maybe = ft->GetFunction();
+  v8::MaybeLocal<v8::Function> maybe = Nan::GetFunction(ft);
   return maybe.ToLocal(out);
 }
 

